snuba/query/functions.py

```diff
@@ -102,3 +102,27 @@
 
 def is_aggregation_function(func_name: str) -> bool:
     return func_name in AGGREGATION_FUNCTIONS
+
+
+# Categorized based on ClickHouse docs
+# https://clickhouse.tech/docs/en/sql-reference/functions/
+REGULAR_FUNCTIONS = {
```

Comment:
not sure yet if this the full list, might be missing some that we use/allow

Reply:
The safest way to verify that the list is complete, is to start the validator in dry-run mode.
This means the valdiator will perform the validation, but, if an invalid function is found, instead of rejecting the query it will log a warning and allow it, so we can find what we missed and fix the list. 
Once we do not log anything anymore we are done and we enable the actual validation.

Reply:
ah yeah okay the wrong part I think I meant `_parse_datetime_literals` maybe? either way, i dont think what I have now validates things like `toDateTime`

Reply:
We should allow type conversion functions. The _parse_datetime_literals is a specific case of `toDateTime`. In SnQL we mark datetime literals by wrapping them in `toDateTime` so we can tell the difference between a string literal that looks like a date and an actual date. Specifically the `tuple` function we use quite a bit.

There are definitely some missing things here. `has`, `if`, `ifNull`, `multiIf`. Take a look at `https://github.com/getsentry/sentry/blob/master/src/sentry/search/events/fields.py#L1584`. These are all the functions that Discover/performance sends. Most of them are aggregate functions but there are other ones as well.